lib/dotloop/loop.rb

```diff
@@ -10,11 +10,12 @@ def initialize(client:)
 
     def all(options = {})
       loops = []
+      options[:batch_size] = BATCH_SIZE
       (1..MAX_LOOPS).each do |i|
         options[:batch_number] = i
-        current_loop = batch(options)
-        loops += current_loop
-        break if current_loop.size < BATCH_SIZE
+        current_batch = batch(options)
+        loops += current_batch
+        break if current_batch.size < BATCH_SIZE
```

Comment:
I see you added line 13, so that now we know the options[:batch_size] is good let's reuse that variable and break on it:

```
break if current_batch.size < options[:batch_size]
```